bld/CLMBuildNamelist.pm

```diff
@@ -2324,6 +2335,12 @@ sub setup_logic_do_transient_pfts {
       $cannot_be_true = "$var cannot be combined with use_fates";
    }
 
+   if (&value_is_true($nl->get_value($var))) {
```

Comment:
Here and the similar new code below: I think this is closer to what's wanted, but I think you do this check too early: at a glance, I think you need this *after* the add_default call a few lines down. Probably the best place for it is at the end of the function.

Reply:
Yes, I see why now. If the functions will switch the do_transient_* options to .false. then the new error checks can pass without aborting and causing the run to fail for no good reason.

Thanks @billsacks

Reply:
And, yes, I agree about waiting to rerun testing until at least a complete review comes through.

Reply:
@slevisconsulting do you have a way to debug the build-namelist changes? You can run build-namelist standalone, so that can help to work on it. Here's something that I do. Setup a case, and then run "preview_namelist -debug" and go back through and get the call to CLM build-namelist from it. You can then mouse it and run it standalone, and change the arguments to do the tests you want to do. The other thing to do is to run it with "perl -d" so it runs in the perl debugger, so you can "s" (step into) "n" (next over) or "c #" (continue to a given line number) to see what's happening.

Reply:
Thank you, @ekluzek , this helps.